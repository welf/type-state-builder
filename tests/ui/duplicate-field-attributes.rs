use type_state_builder::TypeStateBuilder;

#[derive(TypeStateBuilder)]
struct DuplicateAttributes {
    // This should be an error: duplicate attributes
    #[builder(setter_name = "full_name")]
    #[builder(setter_name = "first_name")]
    name: String,

    email: String,
}

fn main() {
    let _instance = DuplicateAttributes::builder()
        .first_name("test".to_string())
        .email("test@example.com".to_string())
        .build();
}
