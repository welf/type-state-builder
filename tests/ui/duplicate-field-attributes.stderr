error: Duplicate setter_name attribute. Only one setter_name is allowed per field
 --> tests/ui/duplicate-field-attributes.rs:7:15
  |
7 |     #[builder(setter_name = "first_name")]
  |               ^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0599]: no function or associated item named `builder` found for struct `DuplicateAttributes` in the current scope
  --> tests/ui/duplicate-field-attributes.rs:14:42
   |
4  | struct DuplicateAttributes {
   | -------------------------- function or associated item `builder` not found for this struct
...
14 |     let _instance = DuplicateAttributes::builder()
   |                                          ^^^^^^^ function or associated item not found in `DuplicateAttributes`
